ngraph/core/src/function.cpp

```diff
@@ -429,18 +430,47 @@ int64_t ov::Function::get_result_index(const Output<Node>& value) const {
     return -1;
 }
 
+namespace {
+class OVTensorWrapper : public ov::runtime::Tensor {
+public:
+    OVTensorWrapper(const std::shared_ptr<ngraph::runtime::HostTensor>& ptr)
+        : ov::runtime::Tensor(ptr->get_element_type(), ptr->get_shape(), ptr->get_data_ptr()),
+          tensor(ptr) {}
+
+private:
+    std::shared_ptr<ngraph::runtime::HostTensor> tensor;
+};
+
+inline ov::runtime::TensorVector copy_tensor(const ngraph::HostTensorVector& tensors) {
+    ov::runtime::TensorVector result;
+    result.reserve(tensors.size());
+    for (const auto& tensor : tensors) {
+        result.emplace_back(std::make_shared<OVTensorWrapper>(tensor));
+    }
+    return std::move(result);
+}
+}  // namespace
+
 bool ov::Function::evaluate(const HostTensorVector& output_tensors,
                             const HostTensorVector& input_tensors,
                             EvaluationContext evaluation_context) const {
+    ov::runtime::TensorVector outputs = copy_tensor(output_tensors);
+    ov::runtime::TensorVector inputs = copy_tensor(input_tensors);
```

Comment:
we can safely pass `ov::runtime::Tensor`s created on top of tmp memory w/o any inheritance, because they will live longer than:
```c++
return evaluate(outputs, inputs, std::move(evaluation_context));
```